document-store/src/test/java/org/hypertrace/core/documentstore/postgres/query/v1/PostgresQueryParserTest.java

```diff
@@ -917,4 +917,35 @@ void testQueryWithFunctionalLhsInRelationalFilter() {
     assertEquals(1, params.getObjectParams().size());
     assertEquals(50, params.getObjectParams().get(1));
   }
+
+  @Test
+  void testQueryQ1DistinctCountAggregationWithMatchingSelectionAndGroupBy() {
+    org.hypertrace.core.documentstore.query.Query query =
+        org.hypertrace.core.documentstore.query.Query.builder()
+            .addSelection(
+                AggregateExpression.of(DISTINCT_COUNT, IdentifierExpression.of("quantity")),
+                "qty_count")
+            .addSelection(IdentifierExpression.of("item"))
+            .addSelection(IdentifierExpression.of("price"))
+            .setFilter(
+                RelationalExpression.of(
+                    IdentifierExpression.of("price"), LTE, ConstantExpression.of(10)))
+            .addAggregation(IdentifierExpression.of("item"))
+            .build();
+
+    PostgresQueryParser postgresQueryParser =
+        new PostgresQueryParser(TEST_COLLECTION, PostgresQueryTransformer.transform(query));
+    String sql = postgresQueryParser.parse();
+
+    assertEquals(
+        "SELECT COUNT(DISTINCT document->>'quantity' ) AS \"qty_count\", "
```

Comment:
It confuses to see where the `price` selection is lost!
Then, I remembered the transformation logic. Anyway, that'll come up.

Reply:
Discussed offline.